File: beecrowd/c++/ad-hoc/1087-queen.cpp
#include <iostream>
#include <cmath>

using namespace std;

int main() {
  double x1, y1, x2, y2;

  while (cin >> x1 >> y1 >> x2 >> y2) {
    if (x1 + y1 + x2 + y2 == 0) {
      break;
    }

    bool diagonal = abs(x1 - x2) == abs(y1 - y2);

    cout << (x1 == x2 && y1 == y2
                 ? "0"
                 : (diagonal || (x1 == x2 || y1 == y2) ? "1" : "2"))
         << endl;
  }

  return 0;
}

File: beecrowd/javascript/ad-hoc/1087-queen.js
const input = require('fs').readFileSync('/dev/stdin', 'utf8');
const lines = input.split('\n');

while (lines.length) {
  const [x1, y1, x2, y2] = lines
    .shift()
    .split(' ')
    .map((e) => Number(e));

  if (x1 + y1 + x2 + y2 === 0) {
    break;
  }

  const diagonal = Math.abs(x1 - x2) == Math.abs(y1 - y2);

  console.log(
    x1 == x2 && y1 == y2 ? '0' : diagonal || x1 == x2 || y1 == y2 ? '1' : '2'
  );
}
